hack/tools/deployment/deployment-update.go

```diff
@@ -0,0 +1,255 @@
+		"osd-fleet-manager-production-ap-southeast-1",
+		"osd-fleet-manager-production-ap-southeast-2",
+		"osd-fleet-manager-production-eu-north-1",
+		"osd-fleet-manager-production-adobe-test",
+		"sadf",
+	}
+
+	ProdGroup3 = []string{
+		"osd-fleet-manager-production-eu-west-2",
+		"osd-fleet-manager-production-eu-central-1",
+		"osd-fleet-manager-production-ca-central-1",
+		"osd-fleet-manager-production-eu-south-1",
+		"osd-fleet-manager-production-us-east-1",
+	}
+
+	ProdGroup4 = []string{
+		"osd-fleet-manager-production-eu-west-1",
+		"osd-fleet-manager-production-us-west-2",
+		"osd-fleet-manager-production-us-east-2",
+		"osd-fleet-manager-production-me-south-1",
+		"osd-fleet-manager-production-sa-east-1",
+		"osd-fleet-manager-production-ap-east-1",
+	}
+
+	ProdGroup5 = []string{
+		"osd-fleet-manager-production-eu-central-2",
+		"osd-fleet-manager-production-eu-south-2",
+		"osd-fleet-manager-production-eu-west-3",
+		"osd-fleet-manager-production-me-central-1",
+	}
+
+	file      = flag.String("file", "", "The path to the deployment specification file.")
+	env       = flag.String("env", "", "The environment to change in the deployment spec.")
+	oldTag    = flag.String("old-tag", "", "The old tag used in search.")
+	newTag    = flag.String("new-tag", "", "The new tag to use.")
+	newCommit = flag.String("new-commit", "", "The new commit SHA to use.")
+)
+
+func main() {
+	// Check all flags were used
+	flag.Parse()
+	if file != nil && len(*file) == 0 ||
+		env != nil && len(*env) == 0 ||
+		oldTag != nil && len(*oldTag) == 0 ||
+		newTag != nil && len(*newTag) == 0 ||
+		newCommit != nil && len(*newCommit) == 0 {
+		log.Fatalf("all flags are required")
+	}
+
+	// Verify the right env value was used
+	if validEnvironment := isValidEnvironment(*env); !validEnvironment {
+		log.Fatalf("invalid environment: %s", *env)
+	}
+	log.Print("Updating deployment spec for environment: ", *env)
+
+	// Read the deployment spec
+	deploymentFileContents, err := os.ReadFile(*file)
+	if err != nil {
+		log.Fatal("failed to read deployment file: ", err)
+	}
+
+	// Split the deployment spec content into a slice of strings
+	lines := strings.Split(string(deploymentFileContents), "\n")
+
+	// Update the deployment spec based on the environment
+	log.Printf("Updating tag from %s to %s and updating SHA to %s", *oldTag, *newTag, *newCommit)
+	err = updateDeploymentByEnvironment(*env, lines, *oldTag, *newTag, *newCommit)
+	if err != nil {
+		log.Fatalf("failed to update deployment: %v", err)
+	}
+
+	// Join the deployment spec back into one file
+	output := strings.Join(lines, "\n")
+
+	// Write the modified contents back to the file
+	err = os.WriteFile(*file, []byte(output), 0644)
+	if err != nil {
+		log.Fatalf("Error: %v", err)
+	}
+
+}
+
+// updateDeploymentByEnvironment updates the sectors in a particular ROSA environment
+func updateDeploymentByEnvironment(env string, lines []string, oldTag, newTag, newCommit string) error {
+	switch env {
```

Comment:
You could factor this to reduce the repetition:

```go
func updateDeploymentByEnvironment(env string, lines []string, oldTag, newTag, newCommit string) error {
    for _, sectors := range map[string][]string{
        "int": Integration,
        // ...
    } {
        for _, sector := range sectors {
            if err := updateTagAndCommitSHA(&lines, sector, oldTag, newTag, newCommit); err != nil {
                return err
            }
        }
    }
}
```

Reply:
The repetition should be gone now.